salvo/src/lib/source_tree.py

```diff
@@ -54,7 +56,11 @@ def __init__(self, source_repo: proto_source.SourceRepository) -> None:
     if not home_dir.startswith(constants.SALVO_TMP):
       home_dir = constants.SALVO_TMP
 
-    self._tempdir = file_ops.get_random_dir(home_dir)
+    # This is the destination directory where the source is copied or cloned
+    # If the source_repo has a path, we copy from that path into the _build_dir.
+    # If the source_repo has a url, we clone from that url into the _build_dir.
+    self._build_dir = file_ops.get_random_dir(home_dir)
```

Comment:
Since we are storing this in an attribute, we aren't really addressing #90, since nothing will destroy the temporary directory after we use it like this. However since this is the build directory we are producing, do we actually want to destroy it after the execution? Or do we want to keep it around for debugging?

If we do want to destroy it, we should ensure that the `__exit__()` method of the created temp directory gets called at some point.

Reply:
I know that when the builder objects go out of scope the temporary directories get cleaned up.   I had to chase a few "File Not Found" errors due to the TemporaryDirectory getting removed.

I know I have to do some work ensuring that various artifacts are cleaned up after benchmarks are executed. We could define `__del__` in the source_tree to ensure that these objects are closed and purged, but that's a separate effort I haven't started digging into yet.

Reply:
Thank you, tracking it as a separate effort is a good idea.